dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/JDBCDecorator.java

```diff
@@ -312,6 +324,60 @@ public long setContextInfo(Connection connection, DBInfo dbInfo) {
     return spanID;
   }
 
+  /**
+   * Executes `SET application_name` statement on the Postgres DB to set the trace parent in
+   * `pg_stat_activity.application_name`. This is used for prepared statements where it isn't
+   * possible to propagate trace parent with the comment. Downside: makes an additional round trip
+   * to the database.
+   *
+   * @param span The span of the instrumented statement
+   * @param connection The same connection as the one that will be used for the actual statement
+   */
+  @SuppressFBWarnings(
+      value = "SQL_NONCONSTANT_STRING_PASSED_TO_EXECUTE",
+      justification = "Prepared statement not possible with SET")
+  public void setApplicationName(AgentSpan span, Connection connection) {
+    final long startTime = System.currentTimeMillis();
+    try {
+
+      Integer priority = span.forceSamplingDecision();
+      if (priority == null) {
+        return;
+      }
+      final String traceParent = DECORATE.traceParent(span, priority);
+      if (traceParent == null || !traceParentPattern.matcher(traceParent).matches()) {
+        throw new IllegalArgumentException("Invalid trace parent: " + traceParent);
+      }
+      final String traceContext = "_DD_" + traceParent;
+
+      // SET doesn't work with parameters
+      final String setCommandBegin = "SET application_name = '";
+      final String setCommandEnd = "';";
+      StringBuilder sql =
+          new StringBuilder(
+              setCommandBegin.length() + traceContext.length() + setCommandEnd.length());
+      sql.append(setCommandBegin);
+      sql.append(traceContext);
+      sql.append(setCommandEnd);
+
+      try (Statement statement = connection.createStatement()) {
+        statement.execute(sql.toString());
```

Comment:
actually the use of a StringBuilder is not really necessary here, since we're not looping on values for instance. We could write this in a much simpler way with no performance impact as
```suggestion
      try (Statement statement = connection.createStatement()) {
        statement.execute("SET application_name = '" + traceContext + "';");
```
since the `+` operator is automatically converted to a StringBuilder by the compiler.